source/extensions/filters/http/ip_tagging/ip_tagging_filter.cc

```diff
@@ -50,16 +122,76 @@ IpTaggingFilterConfig::IpTaggingFilterConfig(
     }
 
     tag_data.emplace_back(ip_tag.ip_tag_name(), cidr_set);
-    stat_name_set_->rememberBuiltin(absl::StrCat(ip_tag.ip_tag_name(), ".hit"));
   }
-  trie_ = std::make_unique<Network::LcTrie::LcTrie<std::string>>(tag_data);
+  return tag_data;
 }
 
-void IpTaggingFilterConfig::incCounter(Stats::StatName name) {
-  Stats::SymbolTable::StoragePtr storage = scope_.symbolTable().join({stats_prefix_, name});
-  scope_.counterFromStatName(Stats::StatName(storage.get())).inc();
+std::shared_ptr<TagSetWatcher>
+TagSetWatcher::create(Server::Configuration::FactoryContext& factory_context, std::string filename,
+                      Stats::Scope& scope, const std::string& stat_prefix) {
+
+  auto ptr =
+      std::make_shared<TagSetWatcher>(factory_context, std::move(filename), scope, stat_prefix);
+  return ptr;
+}
+
+TagSetWatcher::TagSetWatcher(Server::Configuration::FactoryContext& factory_context,
+                             Event::Dispatcher& dispatcher, Api::Api& api, std::string filename,
+                             Stats::Scope& scope, const std::string& stat_prefix)
+    : api_(api), filename_(filename), watcher_(dispatcher.createFilesystemWatcher()), scope_(scope),
+      stat_prefix_(stat_prefix), factory_context_(factory_context),
+      yaml_(absl::EndsWith(filename, MessageUtil::FileExtensions::get().Yaml)) {
```

Comment:
Wonder if we should just try yaml, then json, then give up instead of relying on the extension. but this is probably fine (as long we document it)

Reply:
I did do that before I switched to this version while I was developing. I can change back if you prefer. I think we do either `yaml` or `json` in other part of the code so kept it same. I'm happy to document whichever approach we decide on :)

Reply:
Yeah I don't have a strong feeling here actually (except for needing docs :-) -- so deferring to @htuch since this API-ish related.